src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -69,8 +78,27 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
             _weakETag = weakETag;
             var jobSemaphore = new SemaphoreSlim(1, 1);
 
+            var existingFhirRequestContext = _contextAccessor.FhirRequestContext;
+
             try
             {
+                // Add a request context so Datastore consumption can be added
+                var fhirRequestContext = new FhirRequestContext(
+                    method: OperationsConstants.Reindex,
+                    uriString: "$reindex",
+                    baseUriString: "$reindex",
+                    correlationId: _reindexJobRecord.Id,
+                    requestHeaders: new Dictionary<string, StringValues>(),
+                    responseHeaders: new Dictionary<string, StringValues>())
+                {
+                    IsBackgroundTask = true,
+                    AuditEventType = OperationsConstants.Reindex,
+                };
+
+                _contextAccessor.FhirRequestContext = fhirRequestContext;
+
+                _throttleControllerFactory.Initialize(_reindexJobRecord);
```

Comment:
This usage is a little confusing, IReindexJobThrottleController is a singleton, so it will be called when the reindex job kicks off but sets a globally shared state.

Reply:
Agreed.  I originally made it a singleton to be able to handle RU consumption notifications that are published from another singleton.  Now updated to read the FhirRequestContext.